include/glow/Base/TensorSerialization.h

```diff
@@ -24,35 +24,39 @@
 
 namespace glow {
 
-/// Dump the data content of \p tensor to a raw-binary file without any
-/// other meta information (data type and shape). The binary representation
-/// of data is guaranteed to preserve data precision (bit exactness) upon
-/// round-trips (dump/load).
-void dumpToRawBinaryFile(Tensor &tensor, llvm::StringRef filename);
+/// Dump the content of \p tensor to a binary file \p filename. You can choose
+/// to dump also the tensor type in the file with the flag \p dumpType. The
+/// binary representation of data is guaranteed to preserve data precision
+/// (bit exactness) upon round-trips (dump/load).
+void dumpTensorToBinaryFile(Tensor &tensor, llvm::StringRef filename,
+                            bool dumpType = true);
```

Comment:
Nit: I foresee getting more and more parameters passes to serialization functions in the future and becoming a `SerializationOptions` or `SerializationOptions` ;-) Maybe we should go for it right away and have there only a single field called dumpType for now?

Reply:
Sound reasonable but we have two types of functions: dumping and loading (one has a `dumpType` flag and the other has a `loadType` flag). Should we have two different types of structures (with `dumpType` and `loadType` flags) or just one with (say) a `withType` flag??

Reply:
> Should we have two different types of structures (with dumpType and loadType flags) or just one with (say) a withType flag??

I don't have any strong opinions here. I personally would probably go with `withType` or `isTyped` flag. But I leave up to you to make this call.

Reply:
Created a single `TensorSerializationOptions` structure with the `withType` flag used for both dump/load functions.